### ISSUE TITLE
Export compile commands provider does not resolve header includes

### ISSUE THREAD

**Issue by giraldeau** [15knots/cmake4eclipse#40]
When enabling `CMAKE_EXPORT_COMPILE_COMMANDS Parser`, the headers are not resolved. However, when selecting the `CTD GCC Build Output Parser` and using the GCC compiler, the includes are resolved correctly. I verified that the file `build/Debug/compile_commands.json` exists and its content seems legit: 

```
[
{
  "directory": "/home/francisg/shared/qt5-project/build/Debug",
  "command": "/bin/c++   -DQT_CORE_LIB -DQT_GUI_LIB -DQT_WIDGETS_LIB -I/home/francisg/shared/qt5-project/build/Debug -I/home/francisg/shared/qt5-project -isystem /home/francisg/Qt5.9.0/5.9/gcc_64/include -isystem /home/francisg/Qt5.9.0/5.9/gcc_64/include/QtWidgets -isystem /home/francisg/Qt5.9.0/5.9/gcc_64/include/QtGui -isystem /home/francisg/Qt5.9.0/5.9/gcc_64/include/QtCore -isystem /home/francisg/Qt5.9.0/5.9/gcc_64/./mkspecs/linux-g++  -g   -fPIC -std=gnu++11 -o CMakeFiles/foo.dir/main.cpp.o -c /home/francisg/shared/qt5-project/main.cpp",
  "file": "/home/francisg/shared/qt5-project/main.cpp"
},
...
```

Here is the relevant section from the `language.settings.xml`
```
        <configuration id="cdt.managedbuild.config.gnu.cross.exe.debug.1813080996" name="Debug">
                <extension point="org.eclipse.cdt.core.LanguageSettingsProvider">
                        <provider-reference id="de.marw.cmake.cdt.language.settings.providers.CompileCommandsJsonParser" ref="shared-provider"/>
                        <provider copy-of="extension" id="org.eclipse.cdt.ui.UserLanguageSettingsProvider"/>
                        <provider-reference id="org.eclipse.cdt.core.ReferencedProjectsLanguageSettingsProvider" ref="shared-provider"/>
                        <provider-reference id="org.eclipse.cdt.managedbuilder.core.MBSLanguageSettingsProvider" ref="shared-provider"/>
                        <provider class="org.eclipse.cdt.internal.build.crossgcc.CrossGCCBuiltinSpecsDetector" console="false" env-hash="-308201783778743091" id="org.eclipse.cdt.build.crossgcc.CrossGCCBuiltinSpecsDetector" keep-relative-paths="false" name="CDT Cross GCC Built-in Compiler Settings" parameter="${COMMAND} ${FLAGS} -E -P -v -dD &quot;${INPUTS}&quot;" prefer-non-shared="true">
                                <language-scope id="org.eclipse.cdt.core.gcc"/>
                                <language-scope id="org.eclipse.cdt.core.g++"/>
                        </provider>
                </extension>
        </configuration>
```

P.S I'm sorry for this 3rd issue today ;-)

**Comment by 15knots** [comment]
If you are missing the _include_ folder in the C++ Projects view: CDT takes some time to refresh this view, I could not figure out when (and how to trigger an update). After a restart of Eclipse, these folders will show up.
But the indexer will know of the include directories. You can prove that by placing the cursor in a C file on a macro that is defined inside a header in the said include dir: Press `F3` and eclipse will jump to the macro definition.

**Comment by giraldeau** [comment]
Here is a screenshot obtained after enabling the `CMAKE_EXPORT_COMPILE_COMMANDS` provider and desabling the `CDT GCC Build Output Parser`, force cmake run, clean, build and selecting `Project -> C/C++ Index -> Rebuild`.

![screenshot from 2017-07-12 16-44-13](https://user-images.githubusercontent.com/244101/28139205-5dd179ae-6721-11e7-837f-64e3da581399.png)

Doing the same procedure with `CDT GCC Build Output Parser` enabled, all includes and symbols are resolved correctly.

**Comment by giraldeau** [comment]
I can see that `reducedCommandLine` is cutting more than the tool name. 
Complete command line: `/bin/c++   -DQT_CORE_LIB -DQT_GUI_LIB -DQT_WIDGETS_LIB -I/home/francisg/shared/qt5-project/build/Debug -I/home/francisg/shared/qt5-project -isystem /home/francisg/Qt5.9.1/5.9.1/gcc_64/include -isystem /home/francisg/Qt5.9.1/5.9.1/gcc_64/include/QtWidgets -isystem /home/francisg/Qt5.9.1/5.9.1/gcc_64/include/QtGui -isystem /home/francisg/Qt5.9.1/5.9.1/gcc_64/include/QtCore -isystem /home/francisg/Qt5.9.1/5.9.1/gcc_64/./mkspecs/linux-g++  -g   -fPIC -std=gnu++11 -o CMakeFiles/foo.dir/foo_automoc.cpp.o -c /home/francisg/shared/qt5-project/build/Debug/foo_automoc.cpp`
reducedCommandLine: ` -g   -fPIC -std=gnu++11 -o CMakeFiles/foo.dir/foo_automoc.cpp.o -c /home/francisg/shared/qt5-project/build/Debug/foo_automoc.cpp`

In consequence, no includes are found in `CompileCommandsJsonParser.processCommandLine()`, because they are stripped from the line. By providing the whole line, the argument parser finds the includes, and in the call to `store.addProjectLanguageSettingEntries(project, projectEntries);`, the projectEntries is not empty. However, the includes are still unresolved after running `Project -> Index -> Rebuild`

**Comment by 15knots** [comment]
This was introduced with the fix for #41.
Thanks for pointing me at that!

**Comment by 15knots** [comment]
fixed in 202e4d8fb79c040f61f97008fc8fdc7c0c3e0d9e
